src/rpcnet.cpp

```diff
@@ -398,6 +399,34 @@ static UniValue GetNetworksInfo()
     return networks;
 }
 
+UniValue getdeprecationinfo(const UniValue& params, bool fHelp)
+{
+    if (fHelp || params.size() != 0)
+        throw runtime_error(
+            "getdeprecationinfo\n"
+            "Returns an object containing current version and deprecation block height.\n"
+            "\nResult:\n"
+            "{\n"
+            "  \"version\": xxxxx,                      (numeric) the server version\n"
+            "  \"subversion\": \"/MagicBean:x.y.z[-v]/\",     (string) the server subversion string\n"
+            "  \"deprecationheight\": xxxxx,            (numeric) the deprecation block height\n"
+            "}\n"
+            "\nExamples:\n"
+            + HelpExampleCli("getdeprecationinfo", "")
+            + HelpExampleRpc("getdeprecationinfo", "")
+        );
+
+    LOCK(cs_main);
```

Comment:
No need to take the lock, as this function only depends on hard-coded constants.